frontend/src/BenchmarkTest.res

```diff
@@ -149,23 +143,71 @@ let make = (~data, ~testName, ~comparisonMetrics=?, ~testSelection, ~synchronize
         </tr>
       </thead>
       <tbody>
-        {dataByMetrics
-        ->Belt.Map.String.mapWithKey(renderMetricOverviewRow(~xTicks, ~comparisonMetrics?))
+        {dataByMetricName
+        ->Belt.Map.String.mapWithKey(metricName => {
+          let (comparisonTimeseries, comparisonMetadata) = Belt.Map.String.getWithDefault(
+            comparison,
+            metricName,
+            ([], []),
+          )
+          renderMetricOverviewRow(
+            ~comparison=(comparisonTimeseries, comparisonMetadata),
+            metricName,
+          )
+        })
         ->Belt.Map.String.valuesToArray
         ->Rx.array}
       </tbody>
     </Table>
   }
 
-  let metric_graphs = dataByMetrics
-  ->Belt.Map.String.mapWithKey((metricName, data) => {
+  let metric_graphs = dataByMetricName
+  ->Belt.Map.String.mapWithKey((metricName, (timeseries, metadata)) => {
+    let (comparisonTimeseries, comparisonMetadata) = Belt.Map.String.getWithDefault(
+      comparison,
+      metricName,
+      ([], []),
+    )
+
+    let timeseries = Belt.Array.concat(comparisonTimeseries, timeseries)
+    let metadata = Belt.Array.concat(comparisonMetadata, metadata)
+
+    let xTicks = Belt.Array.reduceWithIndex(timeseries, Belt.Map.Int.empty, (acc, row, index) => {
+      // Use indexed instead of dates. This allows us to map to commits.
+      Belt.Array.set(row, 0, float_of_int(index))->ignore
+      let tick = switch Belt.Array.get(metadata, index) {
+      | Some(xMetadata) =>
+        let xValue = xMetadata["commit"]
+        DataHelpers.trimCommit(xValue)
+      | None => "Unknown"
+      }
+
+      Belt.Map.Int.set(acc, index, tick)
+    })
+
+    let annotations = if Belt.Array.length(comparisonTimeseries) > 0 {
+      let lastComparisonX = Belt.Array.length(comparisonTimeseries) - 1
+      Some([
+        {
+          "series": "value",
+          "x": lastComparisonX,
+          "icon": "branch.svg",
+          "width": 21,
+          "height": 21,
```

Comment:
<img width="159" alt="Screenshot 2021-02-10 at 21 50 58" src="https://user-images.githubusercontent.com/5595092/107570908-6ff86080-6bea-11eb-94ab-90d48374743b.png">

Might be a good idea to set the `text` property too. Don't know what it should represent, but anything else than `undefined` would probably be better 😄

Reply:
Added a label "First PR commit".